File: longestSub.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [1] * (n + 1)
        if n == 1:
            return 1
        
        for i in range(1,n):
            for j in range(i-1,-1,-1):

                if nums[i] > nums[j]:
                    dp[i] = max(dp[i], dp[j]+1)
                    


        val = max(dp)
        return val



        

File: longestSub.rs
impl Solution {
    pub fn length_of_lis(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut dp = vec![1; n+1];

        for i in (1..n){
            for j in (0..i).rev(){
                if nums[j] < nums[i]{
                    dp[i] = std::cmp::max(dp[i], dp[j]+1);
                }
            }
        }
        *dp.iter().max().unwrap() as i32
     
    }
}
